Meijuan Zeng-assignment 2.Rmd

```diff
@@ -0,0 +1,170 @@
+
+g <- graph.data.frame(EDGE, directed=TRUE, vertices=VERTEX)
+
+#Now we can plot our graph using the force directed graphing technique - our old friend Fruchertman-Reingold!
+
+plot(g,layout=layout.fruchterman.reingold)
+
+#There are many ways to change the attributes of the graph to represent different characteristics of the newtork. For example, we can color the nodes according to gender.
+
+plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender)
+
+#We can change the thickness of the edge according to the number of times a particular teacher has sent a particular student to the principal.
+
+plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender, edge.width=EDGE$count)
+````
+
+## Part II
+
+In Part II your task is to [look up](http://igraph.org/r/) in the igraph documentation and create a graph that sizes the student vertices in terms of the number of disciplinary actions they have recieved, and the teachers in terms of the number of disciplinary actions they have given out.
+
+```{r}
+library(tidyr)
+# Disciplinary action count for students
+total_st_EDGE <- EDGE %>% group_by(to) %>% summarize(sum(count)) 
+names(total_st_EDGE) <- c("id","count")
+# Disciplinary action count for teachers
+total_t_EDGE <- EDGE %>% group_by(from) %>% summarize(sum(count))
+names(total_t_EDGE) <- c("id","count")
+total <- bind_rows(total_st_EDGE, total_t_EDGE)
+VERTEX <- full_join(VERTEX, total, by="id")
+# Graph
+my_gr <- graph.data.frame(EDGE, directed=TRUE, vertices=VERTEX)
+plot(my_gr,layout=layout.fruchterman.reingold, 
+     vertex.color=as.factor(VERTEX$gender), 
+     edge.arrow.size = 0.5,
+     edge.width=EDGE$count,
+     vertex.size = VERTEX$count*2)
+```
+
+## Part III
+Now practice with data from our class. Please create a **person-network** with the data set hudk4050-classes.csv. To create this network you will need to create a person-class matrix using the tidyr functions and then create a person-person matrix using `t()`. You will then need to plot a matrix rather than a data frame using igraph.
+
+Once you have done this, also [look up](http://igraph.org/r/) how to generate the following network metrics: betweeness centrality and dregree. **Who is the most central person in the network?**
+
+#data wrangling 
+```{r}
+library(tidyr)
+library(dplyr)
+library(readr)
+
+#Read the file to create d1:
+d1 <- read.csv("HUDK4050-2018-classes.csv", header = TRUE, na.strings = c("", "NA"))
+
+#select name and classes from d1: 
+d2 <- select(d1, "Q8": "Q7") %>% unite(name, Q8, Q9, sep = " ") 
+class_name <- c("Q1 : Q7")
+```
+
+```{r}
+#Gather name by the first and last names,all the classes into class_name, arrange their orders and remove the empty rows:
+d3 <- gather(d2, class, class_name, 2: 7, na.rm = TRUE) %>%  select(name, class_name)%>%arrange(name,class_name)%>%na.omit()%>%filter(class_name>0, class_name != "HUDK 4050")
+
```

Comment:
If we look at the class names in the data set, not all of the formatting is standardized. Some of the class names include spaces and others don't. For example, the code will treat HUDK 4050 and HUDK4050 as two different classes, but they should be the same. You can use gsub to remove the spaces and standardize the format.

Reply:
Hi I just add gsub in my new file.